store/sqlstore/user_store.go

```diff
@@ -576,127 +559,106 @@ func (us SqlUserStore) GetAllProfilesInChannel(channelId string, allowFromCache
+		var users []*model.User
+		if _, err := us.GetReplica().Select(&users, query.String(), query.Args()); err != nil {
 			result.Err = model.NewAppError("SqlUserStore.GetProfilesNotInChannel", "store.sql_user.get_profiles.app_error", nil, err.Error(), http.StatusInternalServerError)
-		} else {
-
-			for _, u := range users {
-				u.Sanitize(map[string]bool{})
-			}
+			return
+		}
 
-			result.Data = users
+		for _, u := range users {
+			u.Sanitize(map[string]bool{})
 		}
+
+		result.Data = users
 	})
 }
 
 func (us SqlUserStore) GetProfilesWithoutTeam(offset int, limit int) store.StoreChannel {
 	return store.Do(func(result *store.StoreResult) {
-		var users []*model.User
+		query := usersQuery.
+			Where(`(
+				SELECT
+					COUNT(0)
+				FROM
+					TeamMembers
+				WHERE
+					TeamMembers.UserId = u.Id
+					AND TeamMembers.DeleteAt = 0
+			) = 0`).
+			OrderBy("u.Username ASC").
+			Offset(offset).Limit(limit)
 
-		query := `
-		SELECT
-			*
-		FROM
-			Users
-		WHERE
-			(SELECT
-				COUNT(0)
-			FROM
-				TeamMembers
-			WHERE
-				TeamMembers.UserId = Users.Id
-				AND TeamMembers.DeleteAt = 0) = 0
-		ORDER BY
-			Username ASC
-		LIMIT
-			:Limit
-		OFFSET
-			:Offset`
-
-		if _, err := us.GetReplica().Select(&users, query, map[string]interface{}{"Offset": offset, "Limit": limit}); err != nil {
+		var users []*model.User
+		if _, err := us.GetReplica().Select(&users, query.String(), query.Args()); err != nil {
 			result.Err = model.NewAppError("SqlUserStore.GetProfilesWithoutTeam", "store.sql_user.get_profiles.app_error", nil, err.Error(), http.StatusInternalServerError)
-		} else {
-
-			for _, u := range users {
-				u.Sanitize(map[string]bool{})
-			}
+			return
+		}
 
-			result.Data = users
+		for _, u := range users {
+			u.Sanitize(map[string]bool{})
 		}
+
+		result.Data = users
 	})
 }
 
 func (us SqlUserStore) GetProfilesByUsernames(usernames []string, teamId string) store.StoreChannel {
 	return store.Do(func(result *store.StoreResult) {
-		var users []*model.User
-		props := make(map[string]interface{})
-		idQuery := ""
+		query := usersQuery
 
-		for index, usernames := range usernames {
-			if len(idQuery) > 0 {
-				idQuery += ", "
-			}
-
-			props["username"+strconv.Itoa(index)] = usernames
-			idQuery += ":username" + strconv.Itoa(index)
+		if teamId != "" {
+			query = query.
+				Join("TeamMembers tm ON (tm.UserId = u.Id AND tm.TeamId = :TeamId)").
+				Bind("TeamId", teamId)
 		}
 
-		var query string
-		if teamId == "" {
-			query = `SELECT * FROM Users WHERE Username IN (` + idQuery + `)`
-		} else {
-			query = `SELECT Users.* FROM Users INNER JOIN TeamMembers ON
-				Users.Id = TeamMembers.UserId AND Users.Username IN (` + idQuery + `) AND TeamMembers.TeamId = :TeamId `
-			props["TeamId"] = teamId
-		}
+		query = query.
+			Where("Username IN (:Usernames)").Bind("Usernames", usernames).
+			OrderBy("u.Username ASC")
```

Comment:
Adding an explicit ordering here to simplify unit testing and remove non-determinism, as above.